network/README.md

```diff
@@ -92,71 +91,53 @@ Some peers aren't discovered through the `PeerList` messages exchanged through p
+A `PeerList` is the message that is used to communicate the presence of peers in the network. Each `PeerList` message contains signed networking-level metadata about a peer that provides the necessary information to connect to it.
 
-Once a `PeerListAck` message is received from a peer, the node that sent the original `PeerList` message marks the corresponding acknowledged validators as already having been transmitted to the peer, so that it's excluded from subsequent iterations of `PeerList` gossip.
+Once peer metadata is received, the node will add that data to its bloom filter to prevent learning about it again.
 
 ###### Gossip
 
 Handshake messages provide a node with some knowledge of peers in the network, but offers no guarantee that learning about a subset of peers from each peer the node connects with will result in the node learning about every peer in the network.
 
-In order to provide a probabilistic guarantee that all peers in the network will eventually learn of one another, each node periodically gossips a sample of the peers that they're aware of to a sample of the peers that they're connected to. Over time, this probabilistically guarantees that every peer will eventually learn of every other peer.
+To provide an eventual guarantee that all peers learn of one another, each node periodically requests peers from a random peer.
 
-To optimize bandwidth usage, each node tracks which peers are guaranteed to know of which peers. A node learns this information by tracking both inbound and outbound `PeerList` gossip.
+To optimize bandwidth, each node tracks the most recent IPs of validators. The validator's nodeID and timestamp are inserted into a bloom filter which is used to select only necessary IPs to gossip.
 
-- Inbound
-  - If a node ever receives `PeerList` from a peer, that peer _must_ have known about the peers in that `PeerList` message in order to have gossiped them.
-- Outbound
-  - If a node sends a `PeerList` to a peer and the peer replies with an `PeerListAck` message, then all peers in the `PeerListAck` must be known by the peer.
+To prevent the bloom filter from having too many false positives, the number of entries a validator is allowed to have in the bloom filter is capped. Periodically, a new bloom filter is generated. Generating the new bloom filter both removes stale entries and modifies the hash functions to avoid persistent hash collisions.
```

Comment:
nit: include that temporary hash collisions can happen where a peer may not be send (as a result of a false positive)